File: subarraySum.py
#User function Template for python3

#Function to find a continuous sub-array which adds up to a given number.
class Solution:
    def subArraySum(self,arr, n, s): 
       #Write your code here
       sum1=0
       l=0
       r=0
       lt=[]
       if s==0:
           lt.append(-1)
           return lt
       while r<n or sum1>=s:
           if sum1<s:
               sum1+=arr[r]
               r+=1
           elif sum1==s:
                lt.append(l+1)
                lt.append(r)
                return lt
           else:
                sum1-=arr[l];
                l+=1
       lt.append(-1)        
       return lt; 


#{ 
 # Driver Code Starts
#Initial Template for Python 3

import math

def main():
        T=int(input())
        while(T>0):
            
            NS=input().strip().split()
            N=int(NS[0])
            S=int(NS[1])
            
            A=list(map(int,input().split()))
            ob=Solution()
            ans=ob.subArraySum(A, N, S)
            
            for i in ans:
                print(i, end=" ")
                
            print()
            
            T-=1


if __name__ == "__main__":
    main()
# } Driver Code Ends

File: Contest116/subarraySum.cpp
//{ Driver Code Starts
//Initial Template for C++
#include<bits/stdc++.h>
using namespace std;

// } Driver Code Ends
//User function Template for C++
class Solution
{
    public:
    long long maxSum(int N, vector<int> A)
    {
        // code here
        long long sum=0;
        // sort(A.begin(),A.end());
        int i=0,count=0;
        while (i<N-1)
        {
            if(A[i]<0)
            count++;
            else
            break;
            A[i]=abs(A[i]);
            i++;
        }
        // if(count%2==1 && i>1)
        // A[i-1]=-A[i-1];
        // sort(A.begin(),A.end());
        for ( i = 0; i < N;i++)
        {
            int l=0,r=0;
            long long tempsum=0;
            while (r<N)
            {
                if(r-l>i){
                    sum+=tempsum;
                    tempsum=tempsum+A[r]-A[l];
                    r++;
                    l++;
                }else{
                    tempsum+=A[r];
                    r++;
                }
            }
            sum+=tempsum;
        }
        

    }
};

//{ Driver Code Starts.
int main()
{

    int t;
    cin>>t;
    while(t--)
    {
        int N;
        cin>>N;
        vector<int> A(N);
        for(int i=0;i<N;i++)
            cin>>A[i];
        Solution obj;
        cout<<obj.maxSum(N, A)<<endl;
    }
}  
// } Driver Code Ends